### PR TITLE
feat: validate signature values

### LINKED ISSUE #1325: bug: AccountContract::validate is not validating the values used to construct the r and s signature values
## Bug Report
It seems AccountContract::validate is not validating the values used to construct the r and s Uint256 values, so the individual legs could be greater than 2^128.
The starknet documentation does not indicate that any constraint is placed on the felts returned by get_tx_info containing the signature. We are not sure whether this can lead to a significant vulnerability or not at the moment, but adding a check seems a good idea regardless.

### Kakarot version
0.8.4

### Expected behavior
Add checks in the validate function

### Related code 
```
let (tx_info) = get_tx_info();

// Assert signature field is of length 5: r_low, r_high, s_low, s_high, v
assert tx_info.signature_len = 5;
let r = Uint256(tx_info.signature[0], tx_info.signature[1]);
let s = Uint256(tx_info.signature[2], tx_info.signature[3]);
```

### PR DESCRIPTION
**Pull request by obatirou**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #1325 

## What is the new behavior?

Signature values are checked to be < 2**128

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1331)

### CODE REVIEW

**Review by enitrat — COMMENTED** [review]
lgtm

File: tests/src/kakarot/accounts/test_account_contract.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from eth_account.account import Account
 from eth_utils import keccak
 from hypothesis import assume, given, settings
-from hypothesis.strategies import binary, composite, integers
+from hypothesis.strategies import binary, composite, integers, sampled_from
```
[the remark this answers is not in the document]
Comment (obatirou) [reply tests/src/kakarot/accounts/test_account_contract.py:10]: nice, did not know that one, thank you !

[the remark this answers is not in the document]
Comment (enitrat) [reply tests/src/kakarot/accounts/test_account_contract.py:10]: my intern that I pay 20$ a month told me about this 😅

File: tests/src/kakarot/accounts/test_account_contract.py [hunk 2]
```diff
@@ -367,6 +367,30 @@ def test_should_raise_with_wrong_signature(self, cairo_run):
                     chain_id=CHAIN_ID,
                 )
 
+        @composite
+        def draw_signature_not_in_range(draw):
+            # create signature with 4 elements < 2 ** 128 and one > 2 ** 128
+            signature = [
+                draw(integers(min_value=0, max_value=2**128 - 1)) for _ in range(4)
```
Comment (enitrat) [inline tests/src/kakarot/accounts/test_account_contract.py:374]: not using `lists` ?

Comment (obatirou) [reply tests/src/kakarot/accounts/test_account_contract.py]: good catch thank you, done with [54fb57e](https://github.com/kkrt-labs/kakarot/pull/1331/commits/54fb57e34b38fd5c2160e97766e3fa076219eb9f)

**Review by enitrat — APPROVED** [review]

**Pull request merged**